synfig-studio/src/gui/canvasview.cpp

```diff
@@ -298,6 +299,21 @@ class studio::CanvasViewUIInterface : public UIInterface
 	virtual bool
 	amount_complete(int current, int total)
 	{
+		float cur_progress=(float)current/(float)total;
```

Comment:
It will be better here if you save the current amount in a private variable and always check whether it has changed before changing the UI elements (this is always slow).

Reply:
not sure what you want here. :`(
I made the float cur_progress private variable.